nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/ConsumeMQTT.java

```diff
@@ -288,9 +289,13 @@ public void process(final OutputStream out) throws IOException {
 
             String transitUri = new StringBuilder(broker).append(mqttMessage.getTopic()).toString();
             session.getProvenanceReporter().receive(messageFlowfile, transitUri);
+            String uuid = messageFlowfile.getAttribute(CoreAttributes.UUID.key());
             session.transfer(messageFlowfile, REL_MESSAGE);
-            mqttQueue.remove(mqttMessage);
             session.commit();
+            if (!mqttQueue.remove(mqttMessage)) {
```

Comment:
In order to avoid concatenating the Strings in the logger message needlessly, this should check if warn is enabled too: 
`... && getLogger().isWarnEnabled()`